test/fixtures/transformation/es6-tail-call/try-catch/expected.js

```diff
@@ -0,0 +1,65 @@
+"use strict";
+
+(function f(n) {
+  if (n <= 0) {
+    return "foo";
+  }
+  try {
+    return f(n - 1);
+  } catch (e) {}
+})(1000000) === "foo";
+
+(function f() {

```

Comment:
I could just not set `node.params = []` and so params will be duplicated in both places, if it's critical. But anyway, I guess I figured out how to implement TCO for general case (not just self-recursion), so please wait with this PR for now.